spec/error_spec.rb

```diff
@@ -4,16 +4,31 @@
 
 describe EasyPost::Error do
   describe 'error' do
-    it 'throws an error on bad shipment data' do
-      expect { EasyPost::Shipment.create }.to raise_error(described_class)
-    end
-
-    it 'converts an error to a string' do
+    it 'handles error parsing' do
       EasyPost::Shipment.create
     rescue described_class => e
-      # rubocop:disable Layout/LineLength
-      expect(e.to_s).to eq('SHIPMENT.INVALID_PARAMS (422): Unable to create shipment, one or more parameters were invalid. [{"to_address"=>"Required and missing."}, {"from_address"=>"Required and missing."}]')
-      # rubocop:enable Layout/LineLength
+      # Test pulling out error properties
+      expect(e.status).to eq(422)
+      expect(e.code).to eq('SHIPMENT.INVALID_PARAMS')
+      expect(e.message).to eq('Unable to create shipment, one or more parameters were invalid.')
+      expect(e.errors[0]).to eq({ 'to_address' => 'Required and missing.' })
+      expect(e.errors[1]).to eq({ 'from_address' => 'Required and missing.' })
+      expect(e.http_body).to eq(
+        '{"error":{"code":"SHIPMENT.INVALID_PARAMS","message":"Unable to create shipment,' \
+        ' one or more parameters were invalid.","errors":[{"to_address":"Required and missing."},' \
+        '{"from_address":"Required and missing."}]}}',
+      )
+
+      # Convert an error to a string
+      expect(e.to_s).to eq(
+        'SHIPMENT.INVALID_PARAMS (422): Unable to create shipment, one or more parameters were invalid.' \
+        ' [{"to_address"=>"Required and missing."}, {"from_address"=>"Required and missing."}]',
+      )
+
+      # Compare an error and its properties to another error
+      # rubocop:disable Lint/BinaryOperatorWithIdenticalOperands
+      expect(e == e).to be true
```

Comment:
You mean `expect(e).to be_equal(e.clone)`.  Comparing an object to itself would pass with the default implementation so cloning is important.

See: https://ruby-doc.org/core-3.1.2/Object.html#method-i-eql-3F

Reply:
This line here is meant to trigger line coverage of our overridden comparison operator for errors: https://github.com/EasyPost/easypost-ruby/blob/083891d5a7d6e792e85efca4ce47bbe5af3f690c/lib/easypost/error.rb#L23. Is there another way to accomplish the same?

Reply:
Yeah use `expect(e).to be_equal(e.clone)` to get the coverage you want with stronger regression protection.

Reply:
Ah, I didn't realize that `eq` would use this under the hood. Thanks!